source/extensions/filters/network/thrift_proxy/router/router_impl.cc

```diff
@@ -502,9 +506,8 @@ void Router::UpstreamRequest::onResetStream(ConnectionPool::PoolFailureReason re
   switch (reason) {
   case ConnectionPool::PoolFailureReason::Overflow:
     parent_.callbacks_->sendLocalReply(
-        AppException(
-            AppExceptionType::InternalError,
-            fmt::format("too many connections to '{}'", upstream_host_->address()->asString())),
+        AppException(AppExceptionType::InternalError,
+                     fmt::format("thrift upstream request: too many connections")),
```

Comment:
Two things: 

1. Since you've removed the string interpolation, you don't need to call `fmt::format` here.
2. There is another case below that uses `upstream_host_` below. That should be modified as well.

Reply:
Do you mean the Timeout case? When it hits that case ,as far as I can see the upstream_host_ will keep the real one.Could you please check if it's right?

Reply:
I don't think we should depend on that behavior even if it's true.

Reply:
I think is better to report more detail about the Exception.How about change like this
`


      parent_.callbacks_->sendLocalReply(
          AppException(
              AppExceptionType::InternalError,
              fmt::format("connection failure '{}'", (upstream_host_ != nullptr)
                                                         ? upstream_host_->address()->asString()
                                                         : "to upstream")),
          true);


`

Reply:
Ok